### PR TITLE
Add org.fcitx.Fcitx5.Addon.M17N

### PR DESCRIPTION
**Pull request by wengxt**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by wengxt** [comment]
bot, build org.fcitx.Fcitx5.Addon.M17N

File: modules/fmt.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: fmt
+buildsystem: cmake-ninja
+config-opts:
+  - -DFMT_TEST=Off
+sources:
+  - type: git
+    url: https://github.com/fmtlib/fmt
+    branch: 7.1.2
```
Comment (hfiguiere) [inline modules/fmt.yaml:8]: you need a tag or a commit, not a branch.

File: org.fcitx.Fcitx5.Addon.M17N.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  prefix: /app/addons/M17N
+  prepend-pkg-config-path: /app/addons/M17N/lib/pkgconfig:/app/addons/M17N/share/pkgconfig
+cleanup:
+  - /bin
+  - /include
+  - /lib/pkgconfig
+  - /share/pkgconfig
+  - '*.la'
+modules:
+  - modules/fmt.yaml
```
Comment (hfiguiere) [inline org.fcitx.Fcitx5.Addon.M17N.yaml:19]: why a separate manifest?

File: org.fcitx.Fcitx5.Addon.M17N.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  - name: m17n-lib
+    make-args:
+      - --jobs=1
```
Comment (hfiguiere) [inline org.fcitx.Fcitx5.Addon.M17N.yaml:22]:
```suggestion
    no-parallel-make: true
```

File: org.fcitx.Fcitx5.Addon.M17N.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+      - type: shell
+        commands:
+          - sed -i "s/freetype-config/pkg-config freetype2/" configure
+  - name: m17n-db
+    sources:
+      - type: archive
+        url: https://download.savannah.gnu.org/releases/m17n/m17n-db-1.8.0.tar.gz
+        sha256: 657f23835b6655e7a63a362bac66260454ee356da4855eadb9366911d33fdc6d
+  - name: fcitx5-m17n
+    buildsystem: cmake
```
Comment (hfiguiere) [inline org.fcitx.Fcitx5.Addon.M17N.yaml:36]: `cmake-ninja` doesn't work?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.fcitx.Fcitx5.Addon.M17N.yaml (hunk 2 above)
Comment (wengxt) [reply org.fcitx.Fcitx5.Addon.M17N.yaml:19]: Basically we maintain a repo and flathub manifest are generated from it and we have our own shared modules https://github.com/fcitx/flatpak-fcitx5/tree/master/modules.. I'd prefer to keep it like this.

Comment (hfiguiere) [reply org.fcitx.Fcitx5.Addon.M17N.yaml:19]: ok

**Comment by nedrichards** [comment]
bot, build org.fcitx.Fcitx5.Addon.M17N

**Pull request closed without merging**